awx/main/utils/plugins.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2024 Ansible, Inc.
+# All Rights Reserved.
+
+"""
+This module contains the code responsible for extracting the lists of dynamically discovered plugins.
+"""
+
+from functools import cache
+
+
+@cache
+def discover_available_cloud_provider_plugin_names() -> list[str]:
+    """Return a list of cloud plugin names available in runtime.
+
+    The discovery result is cached since it does not change throughout
+    the life cycle of the server run.
+
+    :returns: List of plugin cloud names.
+    :rtype: list[str]
+    """
+    from awx.main.models.inventory import InventorySourceOptions
+
+    plugin_names = list(InventorySourceOptions.injectors.keys())
+
+    plugin_names.remove('constructed')
+
+    return plugin_names
+
+
+@cache
+def compute_cloud_inventory_sources() -> dict[str, str]:
+    """Return a dictionary of cloud provider plugin names
+    available plus source control management and constructed.
+
+    :returns: Dictionary of plugin cloud names plus source control.
+    :rtype: dict[str, str]
+    """
+
+    plugins = discover_available_cloud_provider_plugin_names()
+
+    return dict(zip(plugins, plugins), scm='scm', constructed='constructed')
+
+
+@cache
+def return_inventory_source_options() -> dict[str, str]:
```

Comment:
Can we improve the naming here? All functions return something, this word alone does not add any context that wasn't already there. Is it a collection of things for REST API? Why not encode that in the name? What's the relation of options and inventory sources? Does it reference the `OPTIONS` HTTP method or something else?

Reply:
I took a stab at this via this commit: [8f30d5433b9ddf45bffbc01453d7c971ff0fa912](https://github.com/ansible/awx/pull/15537/commits/8f30d5433b9ddf45bffbc01453d7c971ff0fa912) let me know if this works!